api/mesh/v1alpha1/dataplane.proto

```diff
@@ -46,6 +46,10 @@ message Dataplane {
     // Public IP on which the dataplane is accessible in the network.
     string address = 5;
 
+    // AdvertiseAddress is the public routable address for the DP in case
+    // address is not routable; envoy binds to address
```

Comment:
but Envoy binds to the `address` not `advertisedAddress`

Reply:
> Hey, @sudeeptoroy I'm sorry this PR did not land in Kuma 1.2.0. This contribution should have been included.
> I also needed more time to test the PR.
> There is a "typo" in the field and misleading comment but other than that it works as expected.
> 
> We can include this release on release branch and release it in 1.2.1

Thanks for testing this @jakubdyszkiewicz 
I will make the changes you mentioned. 

There is another suggestion i need. 
The code has a function GetIP(). I think this needs to be changed to so that prom can query advertise address if configured. 
This will lead to 3 additional changes and hence the question.

_**1. pkg/envoy/admin/client.go  ---> should this use address or advertise address?**_
2. pkg/mads/util.go. --> this should use advertise address if configured. 
3. pkg/xds/envoy/listeners/v3/access_log_configurer.go  -> this should use advertise address if configured.

Reply:
All of those 3 can use AdvertisedAddress, so we should change GetIP() to return AdvertisedAddress if it is set